test/e2e/specs/fakerp/keyrotation.go

```diff
@@ -53,21 +52,15 @@ var _ = Describe("Key Rotation E2E tests [KeyRotation][Fake][LongRunning]", func
 		mutated := deleteSecrets(internal)
 		Expect(mutated).NotTo(BeNil())
 
-		By("Running generate on the modified config blob")
-		pluginConfig, err := fakerp.GetPluginConfig()
-		Expect(err).NotTo(HaveOccurred())
-		configGen := config.NewSimpleGenerator(pluginConfig)
-		Expect(configGen).NotTo(BeNil())
-		err = configGen.Generate(mutated)
-		Expect(err).NotTo(HaveOccurred())
-
 		By("Persisting the config blob containing the new certificates and credentials")
 		err = saveConfig(mutated, *configBlob)
 		Expect(err).NotTo(HaveOccurred())
 
 		By("Calling update on the fake rp with the updated config blob")
-		updated, err := cli.UpdateOSACluster(ctx, external, pluginConfig)
+		external.Properties.ProvisioningState = nil
+		updated, err := cli.OpenShiftManagedClusters.CreateOrUpdateAndWait(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), *external)
```

Comment:
this should be using `OpenShiftManagedClustersAdmin`

Reply:
Not before https://github.com/openshift/openshift-azure/pull/821 gets merged and also #821 is blocked on this PR so we cannot wait for it. :)